aQute.libg/src/aQute/libg/uri/URIUtil.java

```diff
@@ -41,4 +44,83 @@ public static URI resolve(URI baseURI, String reference) throws URISyntaxExcepti
 		return baseURI.resolve(refURI);
 	}
 
+	private static final URI EMPTYURI = URI.create("");
+
+	/**
+	 * Attempts to fetch a path on the file system for the given URI string.
+	 * Tries a few more tricks than the standard method of
+	 * <tt>Paths.get(new URI(uriString))</tt> - it can handle plain paths, and
+	 * also nested URI pseudo-schemes like <tt>reference:</tt> and
+	 * <tt>jar:</tt>. Examples:
+	 * <ul>
+	 * <li>/path/to/file => /path/to/file</li>
+	 * <li>reference:file:/path/to/file => /path/to/file</li>
+	 * <li>jar:file:/path/to/file.jar!/some/contained/element =>
+	 * /path/to/file.jar</li>
+	 * <li>http://server/path/to/file => <tt>null</tt></li>
+	 * </ul>
+	 *
+	 * @param uriString The string containing the URI for which we are
+	 *            attempting to construct a path.
+	 * @return The method's best guess as to which file on the local filesystem
+	 *         this URI refers, or <tt>null</tt> if it's invalid or not a local
+	 *         filesystem URI.
+	 */
+	public static Path fromURI(String uriString) {
+		if (uriString == null || uriString.isEmpty()) {
+			return null;
+		}
+
+		try {
+			URI uri = resolve(EMPTYURI, uriString);
+			while (uriString != null && !uriString.isEmpty()) {
```

Comment:
How can uriString ever be `null` here? I do see that this is possible.

Reply:
You're right. Originally I didn't have the opening null check in the if statement, so this actually did something.